### PR TITLE
[Fix] FOOTER: Broken URLs

### PR DESCRIPTION
**Pull request by sjokkateer**
Signed-off-by: Remy Bos <27890746+sjokkateer@users.noreply.github.com>

|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description
This PR should fix two broken URLs in the `footer.html`.

### CODE REVIEW

**Comment by froschdesign** [comment]
Looks like we have missed some changes in MkDocs and/or Jinja. 🤦

**Review by froschdesign — APPROVED** [review]

File: theme/partials/footer.html [hunk 1]
```diff
@@ -54,7 +54,7 @@ <h4 class="footer__headline">Support</h4>
                         <h4 class="footer__headline">License</h4>
                         <p>
                             Code licensed under <a
-                                href="{{ config.repo_url }}blob/master/LICENSE.md">BSD 3-Clause</a>.
+                                href="{{ config.repo_url }}/blob/master/LICENSE.md">BSD 3-Clause</a>.
```
Comment (froschdesign) [inline theme/partials/footer.html:57]: This link is wrong because we do not longer use a branch with name `master`.

Comment (froschdesign) [reply theme/partials/footer.html:57]: Related to #61

Comment (froschdesign) [reply theme/partials/footer.html:57]: We need the default branch of the related repository. Example:

https://github.com/laminas/laminas-servicemanager/blob/3.13.x/LICENSE.md

Comment (sjokkateer) [reply theme/partials/footer.html:57]: @froschdesign I see, would the default branch suffice for licensing? If that's the case, it seems that replacing master with HEAD would do the trick?

As mentioned in [SO post](https://stackoverflow.com/a/65158264/13304566), the url could become: https://github.com/laminas/laminas-servicemanager/blob/HEAD/LICENSE.md

Comment (sjokkateer) [reply theme/partials/footer.html:57]: Although for these `/blob/` urls a `-` could be used for it to show the default branch, like:
https://github.com/laminas/laminas-servicemanager/blob/-/LICENSE.md
As mentioned in the comments of previous SO post.

Comment (froschdesign) [reply theme/partials/footer.html:57]: This means we can also fix the problem with edit link.

https://github.com/laminas/documentation-theme/blob/e53c2d3bd381197a6457238a257385d5d8c5d022/theme/partials/content.html#L38-L49

Comment (sjokkateer) [reply theme/partials/footer.html]: @froschdesign I believe these are the changes that should make everything route to the right place again.

File: update_mkdocs_yml.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
     mkdocs = yaml.load(f, Loader=yaml.SafeLoader)
 
 mkdocs["site_url"] = site_url
-mkdocs["edit_uri"] = 'edit/master/{}/'.format(docs_dir)
+mkdocs["edit_uri"] = 'edit/HEAD/{}/'.format(docs_dir)
```
Comment (froschdesign) [inline update_mkdocs_yml.py:18]: Unfortunately this does not work for editing: https://github.com/laminas/laminas-inputfilter/edit/HEAD/docs/book/intro.md

It must be: https://github.com/laminas/laminas-inputfilter/edit/2.19.x/docs/book/intro.md

Comment (sjokkateer) [reply update_mkdocs_yml.py]: @froschdesign  Ok, I undid that commit for now and restored it to fixing the URLs in the footer only.

PR [#61](https://github.com/laminas/documentation-theme/pull/61) seems to be created to deal with the `edit_uri` and specific version branch if I'm right?

**Comment by froschdesign** [comment]
Before version [1.3.0](https://github.com/mkdocs/mkdocs/releases/tag/1.3.0) of MkDocs the `repo_url` was manipulated:

> Setting the edit_uri option no longer erroneously adds a trailing slash to repo_url (https://github.com/mkdocs/mkdocs/pull/2733)

https://github.com/mkdocs/mkdocs/pull/2733/files#diff-685a203254588194fa9c3c97a67d1b8b5609f807541cea23b9f8cd25baa67cec

**Comment by froschdesign** [comment]
@sjokkateer 
Thank you for your time and this contribution! 👍

**Pull request merged**